src/partition/FileUtils.cc

```diff
@@ -82,6 +84,67 @@ void InputFile::read(void *buf, off_t off, size_t sz) const {
     }
 }
 
+InputFileArrow::InputFileArrow(fs::path const &path, off_t blockSize) : InputFile(path), _fd(-1), _sz(-1) {
+    char msg[1024];
+    struct ::stat st;
+
+    m_batchReader = std::make_unique<lsst::partition::ParquetFile>(ParquetFile(path.string().c_str()));
+    arrow::Status status = m_batchReader->SetupBatchReader(blockSize);
+    if (!status.ok()) throw std::runtime_error("Could not setup Arrow recordbatchreader");
+
+    int fd = ::open(path.string().c_str(), O_RDONLY);
+
+    if (fd == -1) {
+        ::strerror_r(errno, msg, sizeof(msg));
+        throw std::runtime_error("open() failed [" + path.string() + "]: " + msg);
+    }
+    if (::fstat(fd, &st) != 0) {
+        ::strerror_r(errno, msg, sizeof(msg));
+        ::close(fd);
+        throw std::runtime_error("fstat() failed [" + path.string() + "]: " + msg);
+    }
+    _fd = fd;
+    _sz = st.st_size;
+}
+
+InputFileArrow::~InputFileArrow() {
+    char msg[1024];
+    if (_fd != -1 && ::close(_fd) != 0) {
+        ::snprintf(msg, sizeof(msg), "close() failed [%s]", _path.string().c_str());
+        ::perror(msg);
+        ::exit(EXIT_FAILURE);
+    }
+}
+
+int InputFileArrow::getBatchNumber() const { return m_batchReader->GetTotalBatchNumber(); }
+
+void InputFileArrow::read(void *buf, off_t off, size_t sz, int *bufferSize,
+                          std::vector<std::string> params) const {
+    char msg[1024];
+    uint8_t *cur = static_cast<uint8_t *>(buf);
+
+    int buffSize;
+    arrow::Status status = m_batchReader->ReadNextBatch_Table2CSV(cur, buffSize, params);
+    // return the buffersize because it varies from batch to batch
+    *bufferSize = buffSize;
```

Comment:
4 lines of the code above would be greatly simplified if you had the input parameter of the method:
```
int *bufferSize
```
declared as a reference:
```
int& bufferSize;
```
In this case, you wouldn't need to introduce an additional local variable `int buffSize;` (which, by the way, is named dangerously to be very close to the input parameter `bufferSize` which complicates visual code inspection). Your code would be simplified to a single line:
```
    arrow::Status status = m_batchReader->ReadNextBatch_Table2CSV(cur, bufferSize, params);
```

Reply:
I switched from pointer to reference.
However the clang formatting tool turned the int& buffersize back into int &buffersize